pkg/signature/algorithm_registry.go

```diff
@@ -222,3 +223,48 @@ func ParseSignatureAlgorithmFlag(flag string) (v1.PublicKeyDetails, error) {
 	}
 	return v1.PublicKeyDetails_PUBLIC_KEY_DETAILS_UNSPECIFIED, fmt.Errorf("could not find matching signature algorithm for flag: %s", flag)
 }
+
+// GetDefaultPublicKeyDetails returns the default public key details for a given key.
+//
+// RSA 2048 => v1.PublicKeyDetails_PKIX_RSA_PKCS1V15_2048_SHA256
+// RSA 3072 => v1.PublicKeyDetails_PKIX_RSA_PKCS1V15_3072_SHA256
+// RSA 4096 => v1.PublicKeyDetails_PKIX_RSA_PKCS1V15_4096_SHA256
+// ECDSA P256 => v1.PublicKeyDetails_PKIX_ECDSA_P256_SHA_256
+// ECDSA P384 => v1.PublicKeyDetails_PKIX_ECDSA_P384_SHA_384
+// ECDSA P521 => v1.PublicKeyDetails_PKIX_ECDSA_P521_SHA_512
+// ED25519 => v1.PublicKeyDetails_PKIX_ED25519_PH
+func GetDefaultPublicKeyDetails(publicKey crypto.PublicKey) (v1.PublicKeyDetails, error) {
```

Comment:
would we want a variant of this function that allows us to pass options, if a client wanted to support PSS instead of PKCS1v1.5 instead? This might also be the answer to ed25519 vs -ph.

Reply:
One thing that might change if we do this is that ed25519ph needs to be enabled with the `WithED25519ph`, so by default with no options you select PureEd25519